src/OpenTelemetry/Trace/Builder/TracerProviderBuilderExtensions.cs

```diff
@@ -24,6 +24,12 @@ public static class TracerProviderBuilderExtensions
     /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/>.</param>
     /// <param name="enabled">Enabled or not. Default value is <c>true</c>.</param>
     /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
+    /// <remarks>
+    /// This method is not supported in native AOT or Mono Runtime as of .NET 8.
+    /// </remarks>
+#if NET7_0_OR_GREATER
+    [RequiresDynamicCode("This method is not supported as of .NET 8.")]
```

Comment:
```suggestion
    [RequiresDynamicCode("The code for detecting exception and setting error status might not be available.")]
```

https://github.com/dotnet/runtime/blob/a2e4177bb55f192483a9f9477a79621338f9c58a/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeGenericTypeParameterBuilder.cs#L71

Reply:
Need to remove ".NET 8" here because if users are using .NET 9 and see such error, they will be super confused (and we don't expect to come back and update this string for each .NET release).